src/test/suite/recorded.test.ts

```diff
@@ -0,0 +1,98 @@
+import * as assert from "assert";
+import { promises as fsp } from "fs";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import * as vscode from "vscode";
+import TestCase, {
+  SerializedPosition,
+  SerializedSelection,
+  TestCaseFixture,
+} from "../../TestCase";
+import { SymbolColor } from "../../constants";
+import { ThatMark } from "../../ThatMark";
+import NavigationMap from "../../NavigationMap";
+import * as sinon from "sinon";
+import { Clipboard } from "../../Clipboard";
+
+function deserializePosition(position: SerializedPosition) {
+  return new vscode.Position(position.line, position.character);
+}
+
+function deserializeSelection(
+  selection: SerializedSelection
+): vscode.Selection {
+  const active = deserializePosition(selection.active);
+  const anchor = deserializePosition(selection.anchor);
+  return new vscode.Selection(anchor, active);
+}
+
+suite("recorded test cases", async function () {
+  const directory = path.join(__dirname, "../../../testFixtures");
+  const files = await fsp.readdir(directory);
+
```

Comment:
Better to drop this here, so that it still tears down mock if test fails

```ts
  teardown(() => {
    sinon.restore();
  });
```

Reply:
I didn't know the name of the hook! Related, do we want to keep mocha in [BDD mode](https://mochajs.org/#interfaces)?  Sort of threw me cause I'm used to the TDD-style hooks and that's what their documentation uses, but also not a big deal.

Reply:
I'm happy to switch to TDD.  Idk why it is in BDD in the template.  The TDD hooks that the documentation suggests silently disable the test if you use them in BDD 😳.  Took me forever to figure that one out haha